nav/backplanes/merge.py

```
@@ -0,0 +1,133 @@
+    if rings_result:
+        ring_arrays: dict[str, Any] = rings_result['arrays']
+        ring_masks: dict[str, Any] = rings_result['masks']
+        ring_distance: np.ndarray = rings_result['distance']
+        ring_name = rings_result['target_key']
+
+        for bp_type in sorted(ring_arrays.keys()):
+            master = np.full((height, width), np.nan, dtype=np.float32)
+            src_vals = ring_arrays[bp_type]
+            src_mask = ring_masks[bp_type]
+            if bp_type not in ring_arrays or bp_type not in ring_masks:
+                raise ValueError(f'Backplane type {bp_type} array or mask not found for rings')
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)